fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/OnlineIndexScrubber.java

```diff
@@ -221,8 +228,21 @@ public Builder setEntriesScanLimit(long entriesScanLimit) {
                 return this;
             }
 
+            /**
+             * Declare that the scrubbed index is a value index, even if it's type's name is different.
+             *
+             * Scrubbing dangling index entries only makes sense for value indexes, where the index entry points back to
+             * the record.
+             * Typically, this function is called to allow scrubbing of an index of a user-defined type. If called, it
+             * is the caller's responsibility to verify that the scrubbed index matches the "value" type criteria.
+             */
+            public Builder declareValueIndex() {
```

Comment:
Rather than `declareValueIndex` or `isValueIndex`, it might be better to call this something like `ignoreIndexTypeCheck`. This is a little bit more forward looking, in that it anticipates a future when we might have more than just `VALUE` indexes allowed by default (there's little reason, for example, that we couldn't add `VERSION` indexes today, for example), and I think it also encapsulates a little better what the affect of setting this flag is.

Reply:
Good point! 
I've eliminated the references to "value" index type in this Javadoc. Trying again..